### ISSUE TITLE
Limit the amount of speculative retries

### ISSUE THREAD

**Issue by dorlaor** [scylladb/scylladb#21214]
Speculative retries are a way to re-run a query automatically in case the current one didn't 
return within p99 of other requests. It's a nice mechanism but can work against you in case the
server is completely loaded. In such a case, all queries will be retried, and the system load
will increase, making the situation worse. 

Suggestion - limit the amount of speculative retry to a budget.
For example, have a leaky bucket measure and cap such retries to
1,000/shard. This wouldn't create a big spike of retries while individual
retries will enjoy speculative execution

**Comment by dani-tweig** [comment]
@mykaul can we reduce the priority? itis not treated as a P1 item.

**Comment by mykaul** [comment]
> [@mykaul](https://github.com/mykaul) can we reduce the priority? itis not treated as a P1 item.

No, it's relevant as it was. It's just very hard to come with a full and correct design for an improvement here.

**Comment by dani-tweig** [comment]
> > [@mykaul](https://github.com/mykaul) can we reduce the priority? itis not treated as a P1 item.
> 
> No, it's relevant as it was. It's just very hard to come with a full and correct design for an improvement here.

@piodul , please provide the steps for mitigation, and an assignee.
